src/averageAges.js

```diff
@@ -36,8 +44,17 @@ function calculateMenAverageAge(people, century) {
  *
  * @return {number}
  */
+
 function calculateWomenAverageAge(people, withChildren) {
-  // write code here
+  const women = people.filter(person => !isMale(person));
+
+  const ages = withChildren
+    ? women
+      .filter(mother => people.some(person => mother.name === person.mother))
+      .map(woman => calculateAge(woman))
+    : women.map(woman => calculateAge(woman));
```

Comment:
You are duplicating this code `.map(woman => calculateAge(woman));` twice. Let's write it once.

Reply:
fixed